### ISSUE TITLE
[Flake] Spanner CreateDatabase takes too long in database_integration_test

### ISSUE THREAD

**Issue by dbolduc** [googleapis/google-cloud-cpp#7510]
GCB: https://console.cloud.google.com/cloud-build/builds;region=us-east1/4342840d-3d82-439a-bcc6-1ad58425e934;tab=detail?project=cloud-cpp-testing-resources
Raw: https://storage.googleapis.com/cloud-cpp-community-publiclogs/logs/google-cloud-cpp/main/7904f0f44ade543207b01eed1d130ce27eaaf58a/fedora-34-integration-daily/log-4342840d-3d82-439a-bcc6-1ad58425e934.txt

The errors look like:
```
[----------] 22 tests from DataTypeIntegrationTest
Creating database and table .......................................................................................................................TIMEOUT
google/cloud/spanner/testing/database_integration_test.cc:132: Failure
Failed
```
... which points to this code: https://github.com/googleapis/google-cloud-cpp/blob/225f0663d17d553d4e68d5f523483e8251ea8bf2/google/cloud/spanner/testing/database_integration_test.cc#L121-L133

These are the affected tests:
```
//google/cloud/spanner/integration_tests:client_integration_test         FAILED in 224.5s
//google/cloud/spanner/integration_tests:data_types_integration_test     FAILED in 225.6s
//google/cloud/spanner/integration_tests:rpc_failure_threshold_integration_test FAILED in 225.6s
//google/cloud/spanner/integration_tests:session_pool_integration_test   FAILED in 221.1s
```

**Comment by devbww** [comment]
https://source.cloud.google.com/results/invocations/4d4d6bd5-c625-45d0-9a67-12e2f94e1dce on macOS.

**Comment by devbww** [comment]
In this last case we saw ...
```
2021-11-01T15:13:51.849045000Z [DEBUG] <0x10cd09dc0> AsyncCreateDatabase(1) <<
2021-11-01T15:19:37.307110000Z [DEBUG] <0x700006971000> AsyncCreateDatabase(1) >>
```
for an elapsed time just to obtain the long-running operation of about 5m45s, so the 5m app-level poll timeout had already expired.

In the previous (non-macOS) cases we saw ...
```
2021-10-29T07:11:27.707399389Z [DEBUG] <139821468375104> AsyncCreateDatabase(1) <<
2021-10-29T07:17:23.584736387Z [DEBUG] <139821442754112> AsyncCreateDatabase(1) >>
```
and
```
2021-10-22T07:19:03.167386218Z [DEBUG] <140261828769856> AsyncCreateDatabase(1) <<
2021-10-22T07:21:23.205296345Z [DEBUG] <140261803148864> AsyncCreateDatabase(1) >>
```
for elapsed times of about 5m54s and 2m20s respectively (the latter being when the app-level poll timeout was 2m).

I wonder how long we would have needed to poll those long-running operations in order to see the final result?  The subsequent `DropDatabase()` calls all succeeded in 5m12s, 4m58s, and 1m45s respectively.  Perhaps we were just seeing general service-latency problems at those moments.

In any case, it seems there is an argument for bumping the app-level poll timeout once more.  Remember that these are tests that will timeout by themselves, and that the default LRO polling limit is 30m, so going from 5m to 10m is reasonable.
